automatic_symbol_loading/automatic_symbol_loading.py

```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
+from typing import Dict
+
+import gdb
+
+try:
+    from elftools.elf.elffile import ELFFile
+except ModuleNotFoundError:
+    check_build_id = False
+else:
+    check_build_id = True
+
+
+def create_file_dict(path: Path) -> dict:
```

Comment:
Sorry, I meant that the return type needs types specified, not `ret_dict`. `ret_dict` needs it only if mypy complains because it cannot infer it.

Without annotation, you have a properly typed dictionary inside the function but, for callers, the dictionary is just a mapping from anything to anything (i.e. `dict` is like saying `Dict[Any, Any]`).

Reply:
Fair enough, I think I didn't think hard enough before fixing it :)